### ISSUE TITLE
Advertisements published on first interface only? 

### ISSUE THREAD

**Issue by svogl** [DeMille/dnssd.js#3]
Hi,
we have  an embedded system with a number of interfaces - eth0, wlan0, bat0 - to name a few.
The following code runs

```
var DNSSD = require("dnssd");

let _dnsdsBrowser = new DNSSD.Browser(DNSSD.tcp('_mqtt'));
let _dnsdsAdvMqtt = new DNSSD.Advertisement(DNSSD.tcp('_funky'), 2883, { name: 'FUNKY' });
let _dnsdsAdvMqttWS = new DNSSD.Advertisement(DNSSD.tcp('_funky-ws'), 3001, { name: 'FUNKY WS' });

_dnsdsAdvMqtt.start();
```
but the services show up only on the first interface that has been detected (eth0 in our case, where we don't want it ... ) -- not even at localhost.

How can we make sure it is accessible from the other network participants?
Simon

**Comment by DeMille** [comment]
Right, I wondered if this might be an issue. It currently binds the socket on `0.0.0.0`, so outgoing packets get sent on whatever interface the kernel thinks is needed to reach the destination. In your case it's probably defaulting to eth0. 

Do you need the advertisements on all the other interfaces or just one, like, wlan0? There's a new method [(setMulticastInterface)](https://nodejs.org/api/dgram.html#dgram_socket_setmulticastinterface_multicastinterface) that was added to node in 8.4 that can set the interface to use for outgoing packets. If all your systems can communicate through one of the interfaces that's the easier answer. (I'm actually pretty excited about that one, they didn't have it when I was trying to figure out how to deal with this before.) 

Otherwise we'll have to create multiple sockets, one for each interface you need to use.

Either way I think I should add an option to specify interfaces somehow.

**Comment by svogl** [comment]
Oh fine - did not notice the multicastInterface call; that could indeed work out as we've set up a bridge interface on all nodes sharing data.
When we publish services will this use the multicastInterface as well?
thanks

**Comment by DeMille** [comment]
Alrighty, I've added that option & published it to npm. Give it a shot and let me know if that works with your network config

**Comment by wasnotrice** [comment]
Hi, I'm running into this issue from the other direction. We've got an app browsing for a service that is advertised by a device connected to local wifi.

- [X] when running on a machine with one network interface, all is well
- [X] when on a machine with multiple interfaces, and they are connected to the same network, all is well
- [ ] when on a machine with multiple interfaces connected to different subnets, then we invariably get the wrong interface.

Is it feasible to pass multiple interfaces to the browser and have it coalesce the results? Ideally, we would be able to discover all available services, regardless of network.

**Comment by sbender9** [comment]
This is an issue for me. I need the advertisement on all the interfaces with the correct ip address. mdns does this correctly,

**Comment by wasnotrice** [comment]
@sbender9 @haakonnessjoen I worked around [my issue](#issuecomment-477185609) by finding all of the active network interfaces, creating one browser instance for each interface, and coalescing results from all browsers. That works fine 👍 

Also see related fix in #16
